pytket/tests/circuit_test.py

```diff
@@ -693,17 +693,96 @@ def test_opgroups() -> None:
 def test_phase_polybox() -> None:
     c = Circuit(1, 1)
     n_qb = 1
-    qubit_indices = {Qubit(0): 1}
-    phase_polynomial = {(True,): 0.1, (False,): 0.1}
-    linear_transformation = np.array([[1, 0], [0, 1]])
+    qubit_indices = {Qubit(0): 0}
+    phase_polynomial = {(True,): 0.1}
+    linear_transformation = np.array([[1]])
+    p_box = PhasePolyBox(n_qb, qubit_indices, phase_polynomial, linear_transformation)
+
+    b = p_box.get_circuit()
+
+    p_box_ii = PhasePolyBox(b)
+
+    c.add_phasepolybox(p_box, [0])
+    c.add_phasepolybox(p_box, [Qubit(0)])
+
+    c.add_phasepolybox(p_box_ii, [0])
+
+    assert p_box.n_qubits == n_qb
+    assert p_box_ii.n_qubits == n_qb
+    assert p_box.qubit_indices == qubit_indices
+    assert p_box_ii.qubit_indices == qubit_indices
+    assert p_box.phase_polynomial == phase_polynomial
+    assert p_box_ii.phase_polynomial == phase_polynomial
+    assert np.array_equal(p_box.linear_transformation, linear_transformation)
+    assert np.array_equal(p_box_ii.linear_transformation, linear_transformation)
+    assert DecomposeBoxes().apply(c)
+
+
+def test_phase_polybox_II() -> None:
+    c = Circuit(1, 1)
+    n_qb = 1
+    qubit_indices = {Qubit(0): 0}
+    phase_polynomial = {(True,): 0.1, (True,): 0.3}
+    linear_transformation = np.array([[1]])
     p_box = PhasePolyBox(n_qb, qubit_indices, phase_polynomial, linear_transformation)
 
+    b = p_box.get_circuit()
+
+    p_box_ii = PhasePolyBox(b)
+
     c.add_phasepolybox(p_box, [0])
     c.add_phasepolybox(p_box, [Qubit(0)])
+
+    c.add_phasepolybox(p_box_ii, [0])
+
     assert p_box.n_qubits == n_qb
+    assert p_box_ii.n_qubits == n_qb
     assert p_box.qubit_indices == qubit_indices
+    assert p_box_ii.qubit_indices == qubit_indices
     assert p_box.phase_polynomial == phase_polynomial
+    assert p_box_ii.phase_polynomial == phase_polynomial
     assert np.array_equal(p_box.linear_transformation, linear_transformation)
+    assert np.array_equal(p_box_ii.linear_transformation, linear_transformation)
+    assert DecomposeBoxes().apply(c)
+
+
+def test_phase_polybox_error() -> None:
+    # uses a unvalid qubit_indices not staring at 0
+    c = Circuit(1, 1)
+    n_qb = 1
+    qubit_indices = {Qubit(0): 1}
+    phase_polynomial = {(True,): 0.1, (True,): 0.3}
+    linear_transformation = np.array([[1]])
+    with pytest.raises(Exception) as e:
```

Comment:
`e` unused (same in two tests below)

Reply:
removed in [](https://github.com/cqc-melf)@cqc-melf
[requsted changes](https://github.com/CQCL/tket/pull/315/commits/9014c6fe28df9e77081f6feeda259fc96444d43a)
[9014c6f](https://github.com/CQCL/tket/pull/315/commits/9014c6fe28df9e77081f6feeda259fc96444d43a)